api/v1alpha1/gatekeeper_types.go

```diff
@@ -140,12 +140,13 @@ const (
 	LogLevelError   LogLevelMode = "ERROR"
 )
 
-// +kubebuilder:validation:Enum:=Enabled;Disabled
+// +kubebuilder:validation:Enum:=Enabled;Disabled;Automatic
 type AuditFromCacheMode string
```

Comment:
Could you please add a comment above `AuditFromCache` in the `AuditConfig` struct to add a description of the field to the CRD? This comment gets automatically inserted into the generated CRD.

Then things like `kubectl explain` will show what `Automatic` means.

Also, please add a note that you can't use `Automatic` if you have referential constraints (i.e. constraints that query for other resources than the resource being audited).